mne/io/snirf/_snirf.py

```diff
@@ -221,6 +222,41 @@ def natural_keys(text):
                                                coord_frame=_frame_to_str[
                                                    coord_frame])
 
+            str_date = np.array((dat.get(
+                '/nirs/metaDataTags/MeasurementDate')))[0].decode('UTF-8')
+            str_time = np.array((dat.get(
+                '/nirs/metaDataTags/MeasurementTime')))[0].decode('UTF-8')
+            str_datetime = str_date + str_time
+
+            meas_date = None
+            # Several formats have been observed so we try each in turn
+            for dt_code in ['%Y-%m-%d%H:%M:%SZ',
+                            '%Y-%m-%d%H:%M:%S']:
+                try:
+                    meas_date = datetime.datetime.strptime(
+                        str_datetime, dt_code)
+                    meas_date = meas_date.replace(tzinfo=datetime.timezone.utc)
+                    break
+                except ValueError:
+                    pass
```

Comment:
Can we be more specific about the error? Could line 238 be placed outside of the try/except block?

Reply:
If the date cant be extracted then its printed in the warning below. Do you think I should throw something in the except also? The dates stored in the files seem to have one of the two formats, so I didn't want errors being thrown all the time (although the first one is the actual correct according to the spec).

If I put the replace outside the try block and the date isn't matched, then it will try and set the timezone for none type. So I am not sure where outside the try to put it.

Reply:
You could `try/except/else`, it makes clearer that the failure is supposed to be just in the `strptime` step

Reply:
Pushed a commit for this plus fix for `flake`